packages/vm/src/instance.rs

```diff
@@ -648,9 +652,12 @@ mod test {
         instance
             .with_querier(|querier| {
                 let response = querier
-                    .handle_query::<Empty>(&QueryRequest::Bank(BankQuery::AllBalances {
-                        address: rich_addr.clone(),
-                    }))
+                    .query::<Empty>(
+                        &QueryRequest::Bank(BankQuery::AllBalances {
+                            address: rich_addr.clone(),
+                        }),
+                        DEFAULT_QUERY_GAS_LIMIT,
```

Comment:
It would be nice to have one test here showing how it handles the OutOfGas from the querier (abort vm).
But I guess that behavior is tested elsewhere?

Reply:
Yeah it should be done somehow, but not easy to test.

`Instance::with_querier` bypasses the contract and executes a query on the VM level. Most likely this (and `with_storage`) should be `pub(crate)` and not be used in integration tests as they consume gas and should only be called from an import implementation.

I'll try to add a test of `do_query_chain` (the import implementation), which would cover this logic:

```rust
let gas_remaining = get_gas_left(ctx);
    let (result, gas_info) = with_querier_from_context::<S, Q, _, _>(ctx, |querier| {
        Ok(querier.query_raw(&request, gas_remaining))
    })?;
```

Reply:
Sounds good. If you can test, great, but the code looks quite clear and minimal and doesn't really need one if it is hard. I will soon have a test in a higher-level, but always good to have tests closer to the code.

Reply:
Hmm, this would not help as well, since it would only test the MockQuerier, not the real one. And for `MockQuerier::query_raw` there is a test already.

I'll merge as-is.